clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -26,13 +26,19 @@
 import java.util.concurrent.atomic.AtomicInteger;
 
 /**
- * The set of requests which have been sent or are being sent but haven't yet received a response
+ * The set of requests which have been sent, or are being sent, but have not yet
+ * received a response. This class is not thread-safe as it pertains to
+ * modification of internal data structures, however, the {@link #count()}
+ * method will return a relatively accurate value even when accessed in a
+ * multi-threaded way.
  */
 final class InFlightRequests {
 
     private final int maxInFlightRequestsPerConnection;
     private final Map<String, Deque<NetworkClient.InFlightRequest>> requests = new HashMap<>();
-    /** Thread safe total number of in flight requests. */
+
+    // Will only be modified by a single thread, but may be accessed by many
+    // threads that wish to read the count
```

Comment:
I think we can be more concise and simply say `// Single writer thread, multiple reader threads`